sdk/sdk-common/src/user/IUser.ts

```diff
@@ -1,6 +1,35 @@
-import { ChainInfo, Wallet } from '@summerfi/sdk-common/common'
+import { ChainInfoDataSchema, IChainInfo } from '../common/interfaces/IChainInfo'
+import { IWallet, WalletDataSchema } from '../common/interfaces/IWallet'
+import { z } from 'zod'
 
-export interface IUser {
-  wallet: Wallet
-  chainInfo: ChainInfo
+/**
+ * Represents a user of the system connected with a wallet on a particular chain
+ */
+export interface IUser extends IUserData {
+  /** The wallet of the user */
+  readonly wallet: IWallet
+  /** The chain the user is connected to */
+  readonly chainInfo: IChainInfo
+}
+
+/**
+ * Zod schema for the data part of IUser
+ */
+export const UserDataSchema = z.object({
+  wallet: WalletDataSchema,
+  chainInfo: ChainInfoDataSchema,
+})
+
+/**
+ * Type for the data part of the IUser interface
+ */
+export type IUserData = Readonly<z.infer<typeof UserDataSchema>>
```

Comment:
This one is really strange. `IUserData` has already wallet and chainInfo. Is this the **hack** that you mentioned to me today?

Reply:
Discussed offline